modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3Service.java

```diff
@@ -254,7 +254,13 @@ protected S3ClientBuilder buildClientBuilder(S3ClientSettings clientSettings, Sd
         }
 
         if (Strings.hasLength(clientSettings.endpoint)) {
-            s3clientBuilder.endpointOverride(URI.create(clientSettings.endpoint));
+            String endpoint = clientSettings.endpoint;
+            if ((endpoint.startsWith("http://") || endpoint.startsWith("https://")) == false) {
+                // Default protocol to https if not specified
+                // See https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/client-configuration.html#client-config-other-diffs
+                endpoint = "https://" + endpoint;
```

Comment:
Another minor comment: It might be useful to have an INFO log here? If the cluster was relying on an explicit `http` protocol setting and default to `https` here would not work for them. In that case, it feels useful to see a log message to indicate this have happened? Since #126843 is a breaking change, I am not entirely sure what the expectation is for end-users. It would be a moot point if they are expected to fix the protocol and endpoint settings before upgrade.

Reply:
That's a good point, I'll add one

Reply:
Added in d3836c4c8f3a0b21a1884296e17d7a7ef58524d8